server/services/rflink/lib/events/rflink.message.js

```diff
@@ -0,0 +1,381 @@
+const logger = require('../../../../utils/logger');
+const RFtoObj = require('../../api/rflink.parse.RFtoObject');
+const { EVENTS, WEBSOCKET_MESSAGE_TYPES } = require('../../../../utils/constants');
+const {
+  DEVICE_FEATURE_CATEGORIES,
+  DEVICE_FEATURE_TYPES,
+  DEVICE_FEATURE_UNITS,
+} = require('../../../../utils/constants');
+
+/**
+ * @description when a message is received by the rflink gateway
+ * @param {string} msgRF - The message.
+ * @example
+ * rflink.message(msg);
+ */
+function message(msgRF) {
+  this.lastCommand = msgRF;
+  this.gladys.event.emit(EVENTS.WEBSOCKET.SEND_ALL, {
+    type: WEBSOCKET_MESSAGE_TYPES.RFLINK.NEW_MESSAGE,
+  });
+  logger.debug(`sending new message ${msgRF}`);
+  const msg = RFtoObj(msgRF);
+  logger.debug(`message RFtoObj => ${JSON.stringify(msg)}`);
+  let newDevice;
+  let doesntExistYet = true;
+
+  if (typeof msg.id === 'string') {
+    if (msg.id.includes('=') === false) {
+      this.newDevices.forEach((d) => {
+        if (`rflink:${msg.id}:${msg.switch}` === d.external_id) {
+          doesntExistYet = false;
+        }
+      });
+
+      if (doesntExistYet === true) {
+        const model = `${msg.protocol.charAt(0).toUpperCase()}${msg.protocol.toLowerCase().slice(1)}`;
+
+        newDevice = {
+          service_id: this.serviceId,
+          name: `${msg.protocol} `,
+          selector: `rflink:${msg.id}:${msg.switch}`,
+          external_id: `rflink:${msg.id}:${msg.switch}`,
```

Comment:
I'm quite sure you can make this file more readable, less if....
Or maybe extract feature creation in specific files...